src/components/nav-bar.jsx

```diff
@@ -42,3 +42,18 @@ function renderUserInfo (user, showLogin, logOut) {
     return <button className="auth-status btn btn-link" onClick={showLogin}>Log In</button>;
   }
 }
+
+function renderNavLink (linkText, filterToMatch, pageFilter, setPageFilter) {
+  const matchesFilter = match => match ? match.url === `/${filterToMatch}` : pageFilter === filterToMatch;
+  const setFilter = () => setPageFilter(filterToMatch);
+
+  return (
+    <NavLink
+      to={`/${filterToMatch}`}
+      isActive={matchesFilter}
+      onClick={setFilter}
```

Comment:
Yup, totally agree that this is wonky, but I think we said that we would be able to handle this better when we moved to a more robust state management system like Redux because I was doing it terribly using `context`! I'll open a ticket as a reminder and merge for now.